js/base/Exchange.js

```diff
@@ -1294,6 +1294,12 @@ module.exports = class Exchange {
             if ((open === undefined) && (last !== undefined) && (change !== undefined)) {
                 open = last - change;
             }
+            if ((vwap !== undefined) && (baseVolume !== undefined) && (quoteVolume === undefined)) {
+                quoteVolume = vwap / baseVolume;
```

Comment:
what's the reason to use vwap for the emulated volumes, and not last (or open or close or ...)?

If vwap is not provided by the exchange (which it isn't in case of okx futures) - then it's emulated as `baseVolume / quoteVolume` (which is not the correct calculation for vwap based on the original vwap definition, but is a different topic altogether - so please ignore this for now).

This however leads to the fact that if either `baseVolume` or `quoteVolume` and `vwap` is missing - the emulation is not emulating (not for quoteVolume, not for baseVolume).

Maybe a fallback (use vwap or last?) might be an option to make this a bit more reliable?

Reply:
@xmatthias using `last`, `open`, `close` for this calculation is not entirely correct, cause those are prices at specific fixed moments in time, while vwap is a volume-weighed average price, so using `vwap` for this calculation is better than using any other price value. In this case, however, we cannot calculate it at all, cause this exchange does not specify `vwap` or `quoteVolume`.